### ISSUE TITLE
[bug]preprocessor_definitions.Release["XXX"] = None generate not corrext

### ISSUE THREAD

**Issue by peterpuppy** [conan-io/conan#15753]
### Environment details

* Operating System+version:windows11
* Compiler+version: vc142
* Conan version: 2.1
* Python version:3.8
tc.preprocessor_definitions[f"PROFILE"] = None

### Steps to reproduce

in conanfile.py
```
tc.preprocessor_definitions[f"PROFILE"] = None
```
conan install in toolchain file 

```
# Preprocessor definitions per configuration
set(CONAN_DEF_ReleaseRelease "None")
add_compile_definitions(Release=$<IF:$<CONFIG:Release>,${CONAN_DEF_ReleaseRelease},"">)
```

**Comment by memsharded** [comment]
Hi @peterpuppy 

Thanks for your report.

We have a test in our suite that tests this in https://github.com/conan-io/conan/blob/2cf7c3258abd0614f85f683c1564efcb91181410/conans/test/functional/toolchains/cmake/test_cmake_toolchain.py#L341

It seems it is generating the correct toolchain.cmake code:
```cmake
add_compile_definitions("NOVALUE_DEF")
```

And the C++ code is correctly using it:
```cpp
#ifdef NOVALUE_DEF
            printf("NO VALUE!!!!");
#endif
```

So I'd say this is working fine.

**Comment by memsharded** [comment]
It seems that the issue could be related to using ``tc.preprocessor_definitions.release[f"PROFILE"] = None``, with using the ``.<config>``. I have doing some improvements in https://github.com/conan-io/conan/pull/15756 that would fix this issue, and it will be included in next 2.2 release

**Comment by peterpuppy** [comment]
yes it's.
```
tc.preprocessor_definitions.release[f"PROFILE"] = None
```
my mistake
